### PR TITLE
Pwm color sensor

### PR DESCRIPTION
**Pull request by Maanav-s**

### CODE REVIEW

**Comment by Maanav-s** [comment]
Have one single `m_output` instead of `m_red` `m_blue` etc.

**Comment by Maanav-s** [comment]
Have the DIO signals send in the color select methods instead of doing IO in periodic

**Comment by Maanav-s** [comment]
Remove Sendables so we can move it to Sushi Lib

File: src/main/java/frc/robot/subsystems/TCS3200_ColorSensor.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    ColorSelect(int i) {
+      val = i;
+    }
+  }
+
+  private double m_red;
+  private double m_green;
+  private double m_blue;
+  private double m_clear;
+
```
Comment (Maanav-s) [inline src/main/java/frc/robot/subsystems/TCS3200_ColorSensor.java:26]: The if statements cannot be the most efficient way to structure this, we also want default values for all of them

File: src/main/java/frc/robot/subsystems/TCS3200_ColorSensor.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      m_s1 = new DigitalOutput(s1);
+    }
+
+    if (s2 != -1) {
+      m_s2 = new DigitalOutput(s2);
+    }
+
+    if (s3 != -1) {
+      m_s3 = new DigitalOutput(s3);
+    }
```
Comment (Maanav-s) [inline src/main/java/frc/robot/subsystems/TCS3200_ColorSensor.java]: The if statements cannot be the most efficient way to structure this, we also want default values for all of them

**Comment by Maanav-s** [comment]
After testing integration remove `extends SubsystemBase` and `@Override`s and move to Sushi Lib. Will need to call this periodic in the subsystem periodic

**Comment by loukylor** [comment]
> Have one single `m_output` instead of `m_red` `m_blue` etc.

I should probably do this, however having `m_red`, etc. makes less boiler plate code.

File: src/main/java/frc/robot/subsystems/TCS3200_ColorSensor.java (hunk 2 above)
Comment (loukylor) [reply src/main/java/frc/robot/subsystems/TCS3200_ColorSensor.java]: resolved in e48b7db5abfdf7b56b1083fc0df72ed9ebe5214f

**Comment by loukylor** [comment]
> Have the DIO signals send in the color select methods instead of doing IO in periodic

Not sure what you mean by this. There is no reading or writing of any DIO ports being done in periodic. The counter's `get()` and `reset()` methods are the only thing of significance that is run in periodic.

File: src/main/java/frc/robot/subsystems/TCS3200_ColorSensor.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+    double time = Timer.getFPGATimestamp();
+
+    m_red = 0;
+    m_blue = 0;
+    m_green = 0;
+    m_clear = 0;
+
+    // Scale the number counted to amount of time that passed since last check
+    double freq = (m_counter.get() / (time - m_lastTimeStamp));
```
Comment (loukylor) [inline src/main/java/frc/robot/subsystems/TCS3200_ColorSensor.java]: We should consider replacing `get()` with `getRate()`

**Comment by Maanav-s** [comment]
@loukylor oh i see i think i was being a clown. This makes sense and the new commit looks good

**Comment by Maanav-s** [comment]
Are we ready to move this or do we wait for testing on robot

**Comment by loukylor** [comment]
> Are we ready to move this or do we wait for testing on robot

It should work, however we will have to change how we use the color sensor on the indexer branch.

**Comment by loukylor** [comment]
Moved to SushiFrcLib.

**Pull request closed without merging**